TAA rework

Zvork opened the pull request:
This PR is a rework of the Temporal Anti-Aliasing algorithm to try to respond to [user issues expressed in RC67](https://highfidelity.fogbugz.com/f/cases/15110/Users-reporting-blurring-issues-with-the-new-TAA-feature-desktop-and-HMD). The rework consits in:

1. Moving the jittering to the present thread to have a higher frequency jittering,
2. Taking into account object movement and not just camera movement. This implies writing the velocity buffer during the G-Buffer writing phase and each entity class to keep track of its previous model transform.
3. There is also a slight change in how we unjitter that produces a sharper final result.

In the current state:

- transparent objects are jittered but they don't write to the velocity buffer and thus are not directly anti-aliased. It should work most of the time, though.
- background is anti-aliased and writes to velocity buffer.
- skinned (deformed) models and a number of entities don't track their previous models so their velocity relies only on camera movement.

Hifi_tests have a PR that updates the autoTester.js with this new PR: https://github.com/highfidelity/hifi_tests/pull/280

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10146/HighFidelity-Beta-PR13246-32f31dceccdcacde1e9f98a32660519296503542.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10146/HighFidelity-Beta-Interface-PR13246-32f31dceccdcacde1e9f98a32660519296503542.exe

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10147/HighFidelity-Beta-PR13246-a710821bd17b066c19c0d61faea1ecfa972d31a4.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10147/HighFidelity-Beta-Interface-PR13246-a710821bd17b066c19c0d61faea1ecfa972d31a4.exe

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10148/HighFidelity-Beta-PR13246-c4f91c58d02222c3170d04eb8b10e6bd210dba87.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10148/HighFidelity-Beta-Interface-PR13246-c4f91c58d02222c3170d04eb8b10e6bd210dba87.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10149/HighFidelity-Beta-PR13246-72e12a1f6b6e5255cc71dc35ebb09b6e6332d319.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10149/HighFidelity-Beta-Interface-PR13246-72e12a1f6b6e5255cc71dc35ebb09b6e6332d319.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10149/HighFidelity-Beta-PR13246-72e12a1f6b6e5255cc71dc35ebb09b6e6332d319.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10149/HighFidelity-Beta-Interface-PR13246-72e12a1f6b6e5255cc71dc35ebb09b6e6332d319.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10159/HighFidelity-Beta-PR13246-255d6bc1bd40c22310d646d4929b02827dca1cad.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10159/HighFidelity-Beta-Interface-PR13246-255d6bc1bd40c22310d646d4929b02827dca1cad.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10159/HighFidelity-Beta-PR13246-255d6bc1bd40c22310d646d4929b02827dca1cad.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10159/HighFidelity-Beta-Interface-PR13246-255d6bc1bd40c22310d646d4929b02827dca1cad.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10160/HighFidelity-Beta-PR13246-896d0aa3f697ade8a9f64087c224eed00d27b899.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10160/HighFidelity-Beta-PR13246-896d0aa3f697ade8a9f64087c224eed00d27b899.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10160/HighFidelity-Beta-Interface-PR13246-896d0aa3f697ade8a9f64087c224eed00d27b899.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10160/HighFidelity-Beta-PR13246-896d0aa3f697ade8a9f64087c224eed00d27b899.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10160/HighFidelity-Beta-Interface-PR13246-896d0aa3f697ade8a9f64087c224eed00d27b899.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10161/HighFidelity-Beta-PR13246-555d0fd63471fcee539a02ec1aaed1f737994ff6.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10161/HighFidelity-Beta-PR13246-555d0fd63471fcee539a02ec1aaed1f737994ff6.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10161/HighFidelity-Beta-Interface-PR13246-555d0fd63471fcee539a02ec1aaed1f737994ff6.dmg

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10161/HighFidelity-Beta-Interface-PR13246-555d0fd63471fcee539a02ec1aaed1f737994ff6.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10162/HighFidelity-Beta-PR13246-47280c28eec13c08a43977b847f42076734e074c.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10162/HighFidelity-Beta-PR13246-47280c28eec13c08a43977b847f42076734e074c.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10162/HighFidelity-Beta-Interface-PR13246-47280c28eec13c08a43977b847f42076734e074c.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10162/HighFidelity-Beta-PR13246-47280c28eec13c08a43977b847f42076734e074c.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10162/HighFidelity-Beta-Interface-PR13246-47280c28eec13c08a43977b847f42076734e074c.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10167/HighFidelity-Beta-PR13246-392bffebf11aa0baa7661627d27fa2316bd50dc4.apk

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10167/HighFidelity-Beta-Interface-PR13246-392bffebf11aa0baa7661627d27fa2316bd50dc4.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10205/HighFidelity-Beta-PR13246-1f3da7d9032ebcbf9b3c3df9500bf8fa029edafd.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10205/HighFidelity-Beta-PR13246-1f3da7d9032ebcbf9b3c3df9500bf8fa029edafd.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10205/HighFidelity-Beta-Interface-PR13246-1f3da7d9032ebcbf9b3c3df9500bf8fa029edafd.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10205/HighFidelity-Beta-PR13246-1f3da7d9032ebcbf9b3c3df9500bf8fa029edafd.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10205/HighFidelity-Beta-Interface-PR13246-1f3da7d9032ebcbf9b3c3df9500bf8fa029edafd.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10263/HighFidelity-Beta-PR13246-e8fe7de.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10263/HighFidelity-Beta-PR13246-e8fe7de.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10263/HighFidelity-Beta-Interface-PR13246-e8fe7de.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10263/HighFidelity-Beta-PR13246-e8fe7de.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10263/HighFidelity-Beta-Interface-PR13246-e8fe7de.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10856/HighFidelity-Beta-PR13246-5b556ff.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10856/HighFidelity-Beta-PR13246-5b556ff.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10856/HighFidelity-Beta-Interface-PR13246-5b556ff.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10856/HighFidelity-Beta-PR13246-5b556ff.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10856/HighFidelity-Beta-Interface-PR13246-5b556ff.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10857/HighFidelity-Beta-PR13246-d0d013c.apk

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10857/HighFidelity-Beta-PR13246-d0d013c.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10857/HighFidelity-Beta-Interface-PR13246-d0d013c.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10971/HighFidelity-Beta-PR13246-ed62e1b.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10971/HighFidelity-Beta-PR13246-ed62e1b.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10971/HighFidelity-Beta-Interface-PR13246-ed62e1b.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10971/HighFidelity-Beta-PR13246-ed62e1b.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10971/HighFidelity-Beta-Interface-PR13246-ed62e1b.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11079/HighFidelity-Beta-PR13246-3098bed.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11079/HighFidelity-Beta-Interface-PR13246-3098bed.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11079/HighFidelity-Beta-PR13246-3098bed.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11079/HighFidelity-Beta-Interface-PR13246-3098bed.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11080/HighFidelity-Beta-PR13246-9395ad6.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11080/HighFidelity-Beta-PR13246-9395ad6.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11080/HighFidelity-Beta-Interface-PR13246-9395ad6.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11080/HighFidelity-Beta-PR13246-9395ad6.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11080/HighFidelity-Beta-Interface-PR13246-9395ad6.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11190/HighFidelity-Beta-PR13246-d87edc6.apk

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11190/HighFidelity-Beta-PR13246-d87edc6.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11190/HighFidelity-Beta-Interface-PR13246-d87edc6.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11268/HighFidelity-Beta-PR13246-fc527d9.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11268/HighFidelity-Beta-Interface-PR13246-fc527d9.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11268/HighFidelity-Beta-PR13246-fc527d9.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11268/HighFidelity-Beta-Interface-PR13246-fc527d9.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11415/HighFidelity-Beta-PR13246-c692669.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11415/HighFidelity-Beta-PR13246-c692669.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11415/HighFidelity-Beta-Interface-PR13246-c692669.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11415/HighFidelity-Beta-PR13246-c692669.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11415/HighFidelity-Beta-Interface-PR13246-c692669.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11969/HighFidelity-Beta-PR13246-aa547cc.apk

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11970/HighFidelity-Beta-PR13246-64655d7.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11970/HighFidelity-Beta-PR13246-64655d7.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11970/HighFidelity-Beta-Interface-PR13246-64655d7.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11970/HighFidelity-Beta-PR13246-64655d7.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11970/HighFidelity-Beta-Interface-PR13246-64655d7.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12662/HighFidelity-Beta-PR13246-3bd0740.apk

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12662/HighFidelity-Beta-PR13246-3bd0740.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12662/HighFidelity-Beta-Interface-PR13246-3bd0740.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12701/HighFidelity-Beta-PR13246-1e8af9b.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12701/HighFidelity-Beta-PR13246-1e8af9b.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12701/HighFidelity-Beta-Interface-PR13246-1e8af9b.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12701/HighFidelity-Beta-PR13246-1e8af9b.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12702/HighFidelity-Beta-PR13246-8e2a8b6.apk

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12702/HighFidelity-Beta-Interface-PR13246-8e2a8b6.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12853/HighFidelity-Beta-PR13246-bb8dae3.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12853/HighFidelity-Beta-Interface-PR13246-bb8dae3.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12853/HighFidelity-Beta-PR13246-bb8dae3.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12853/HighFidelity-Beta-Interface-PR13246-bb8dae3.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13061/HighFidelity-Beta-PR13246-be96d25.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13061/HighFidelity-Beta-Interface-PR13246-be96d25.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13061/HighFidelity-Beta-PR13246-be96d25.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13061/HighFidelity-Beta-Interface-PR13246-be96d25.exe

Zvork commented:
build this please

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13062/HighFidelity-Beta-PR13246-be96d25.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13062/HighFidelity-Beta-Interface-PR13246-be96d25.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13062/HighFidelity-Beta-PR13246-be96d25.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13062/HighFidelity-Beta-Interface-PR13246-be96d25.exe

Zvork commented:
build this please

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13063/HighFidelity-Beta-PR13246-be96d25.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13063/HighFidelity-Beta-Interface-PR13246-be96d25.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13063/HighFidelity-Beta-PR13246-be96d25.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13063/HighFidelity-Beta-Interface-PR13246-be96d25.exe

samcake commented:
build this please

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13300/HighFidelity-Beta-PR13246-be96d25.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13300/HighFidelity-Beta-PR13246-be96d25.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13300/HighFidelity-Beta-Interface-PR13246-be96d25.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13300/HighFidelity-Beta-PR13246-be96d25.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13300/HighFidelity-Beta-Interface-PR13246-be96d25.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13743/HighFidelity-Beta-PR13246-ec0833b.apk

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13744/HighFidelity-Beta-PR13246-eec5588.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13744/HighFidelity-Beta-PR13246-eec5588.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13744/HighFidelity-Beta-Interface-PR13246-eec5588.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13744/HighFidelity-Beta-PR13246-eec5588.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13744/HighFidelity-Beta-Interface-PR13246-eec5588.exe

samcake commented on libraries/render-utils/src/Haze.slf:
```diff
@@ -44,9 +43,10 @@ void main(void) {
+    vec4 fragPositionES = unpackPositionFromZeyeAndGetSide(varTexCoord0, side);
 
-    mat4 viewInverse = getViewInverse();
+    mat4 viewInverse = getViewInverse(side);
```
THis looks like it was a bug isnt it ?

samcake replied:
got  it

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14005/HighFidelity-Beta-PR13246-8856232.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14005/HighFidelity-Beta-PR13246-8856232.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14005/HighFidelity-Beta-Interface-PR13246-8856232.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14005/HighFidelity-Beta-PR13246-8856232.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14005/HighFidelity-Beta-Interface-PR13246-8856232.exe

SamGondelman commented on libraries/render-utils/src/taa.slh:
```diff
@@ -76,19 +81,39 @@ vec2 taa_getRegionFXAA() {
+}
+
+vec4 taa_transformColor(vec4 c) {
+	// Apply rapid pseudo tonemapping as TAA is applied to a tonemapped image, using luminance as weight, as proposed in
```
buncha tabs in this file

Zvork replied:
Yep

SamGondelman commented on libraries/render-utils/src/taa.slh:
```diff
@@ -105,7 +130,50 @@ vec4 taa_fetchSourceMap(vec2 uv) {
+        // Perform a Catmull-Rom interpolation for a (possibly) sharper result.
+        // Optimized 9-tap fetch based on http://vec3.ca/bicubic-filtering-in-fewer-taps/
+        vec2 samplePixelPos = uv * taa_getImageSize();
+        vec2 samplePixelIntPos = floor(samplePixelPos - 0.5) + 0.5;
```
Is this doing what you want?  The -0.5 will give you the pixel before, right?

Zvork replied:
Well the formula in GL to go from pixel to UV is uv = (pixel + 0.5) / dim so, the inverse is pixel = uv ù dim - 0.5
The error seems to be the +0.5 and that's because I stupidly copied a DirectX code :(
I've just fixed things in the shader and now the bicubic sampling gives a slighly sharper result than the traditionnal bilinear fetch. This wasn't the case before as there was hardly any difference.
I would even recommend enabling bicubic fetch and putting the post sharpen to 0. But I'll leave that to the artists to decide...

SamGondelman commented on libraries/render/src/render/ShapePipeline.cpp:
```diff
@@ -100,9 +100,9 @@ void ShapePlumber::addPipeline(const Filter& filter, const gpu::ShaderPointer& p
+    locations->deferredFrameTransformBufferUnit = program->getUniformBuffers().isValid(render_utils::slot::buffer::DeferredFrameTransform);
     locations->lightAmbientMapUnit = program->getTextures().isValid(graphics::slot::texture::Skybox);
     locations->fadeMaskTextureUnit = program->getTextures().isValid(render_utils::slot::texture::FadeMask);
-    locations->fadeParameterBufferUnit = program->getUniformBuffers().isValid(render_utils::slot::buffer::FadeParameters);
```
Why'd you get rid of this?

Zvork replied:
I have no idea... I'll put it back. I really don't understand what all these flags are for as they same completely unused, by the way.

SamGondelman commented on scripts/developer/utilities/render/deferredLighting.qml:
```diff
@@ -43,13 +43,14 @@ Rectangle {
            // padding: 10
                 Repeater {
                     model: [
+                         "Opaque Overlays:DrawOverlayInFrontOpaque:enabled",
```
Hm what does this do?

Zvork replied:
Nothing :-). At some point I wanted a quick way to disable the overlay drawing to quickly determine what objects where "overlays" but, as it turns out, simply disabling the DrawOverlayInFrontOpaque job doesn't work. I think I talked to Sam about this and then completely forgot about it.
I'll remove it.

Zvork commented on libraries/gpu-gl-common/src/gpu/gl/GLBackend.cpp:
```diff
@@ -422,8 +424,12 @@ void GLBackend::render(const Batch& batch) {
+    // Update object transform history for when the batch will be reexecuted
+    for (auto& objectTransform : batch._objects) {
+        objectTransform._previousModel = objectTransform._model;
+    }
```
I don't really see how doing this in the renderPassDraw function will change this? Do you have something in mind?
But what we could do is have some sort of "mutable" boolean flag on the batch to know if we have to redo if the batch is rendered a second (or more) time. This boolean flag would be see just after this loop.

Zvork commented on libraries/gpu-gl-common/src/gpu/gl/GLBackend.cpp:
```diff
@@ -742,8 +741,7 @@ void GLBackend::queueLambda(const std::function<void()> lambda) const {
 void GLBackend::recycle() const {
-    PROFILE_RANGE(render_gpu_gl, __FUNCTION__)
-    {
+    PROFILE_RANGE(render_gpu_gl, __FUNCTION__) {
```
done

Zvork commented on libraries/gpu-gl-common/src/gpu/gl/GLBackend.cpp:
```diff
@@ -49,9 +48,15 @@ GLBackend::CommandCall GLBackend::_commandCalls[Batch::NUM_COMMANDS] =
     (&::gpu::gl::GLBackend::do_setViewTransform),
     (&::gpu::gl::GLBackend::do_setProjectionTransform),
     (&::gpu::gl::GLBackend::do_setProjectionJitter),
```
True. Fixed

Zvork commented on libraries/render-utils/src/surfaceGeometry_makeCurvature.slf:
```diff
@@ -141,10 +139,10 @@ void main(void) {
-    vec3 ax = (frameTransform._view[0].xyz * axeLength);
-    vec3 ay = (frameTransform._view[1].xyz * axeLength);
-    vec3 az = (frameTransform._view[2].xyz * axeLength);
+	mat4 view = getView(stereoSide.x);
```
Fixed

Zvork commented on libraries/gpu-gl-common/src/gpu/gl/GLBackend.cpp:
```diff
@@ -861,17 +859,23 @@ void GLBackend::recycle() const {
+    _transform._presentFrame.correctionInverse = invCorrection;
+
+    auto& projectionJitter = _transform._projectionJitter;
+    projectionJitter._currentSampleIndex++;
```
Yes, because the jitter offset sequence can be set sometimes during the batch, later on. It is not necessarilly set at the start of the batch so still need to keep it incrementing, in case there is a jitter sequence used elsewhere. See do_setProjectionJitterSequence.
The modulo operator should be different, now that I think of it. If two different sequences are set during the batch, with varying lenghts (it is theoritically possible even though it might not be of a great interest), the first sequence length will be used as the modulo... Not great. I'll fix this.

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14082/HighFidelity-Beta-PR13246-4cfc874.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14082/HighFidelity-Beta-PR13246-4cfc874.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14082/HighFidelity-Beta-Interface-PR13246-4cfc874.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14082/HighFidelity-Beta-PR13246-4cfc874.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14083/HighFidelity-Beta-PR13246-a0a91ec.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14083/HighFidelity-Beta-Interface-PR13246-a0a91ec.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14083/HighFidelity-Beta-PR13246-a0a91ec.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14083/HighFidelity-Beta-Interface-PR13246-a0a91ec.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14084/HighFidelity-Beta-PR13246-c0254dc.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14084/HighFidelity-Beta-Interface-PR13246-c0254dc.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14084/HighFidelity-Beta-PR13246-c0254dc.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14084/HighFidelity-Beta-Interface-PR13246-c0254dc.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14089/HighFidelity-Beta-PR13246-a482283.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14089/HighFidelity-Beta-PR13246-a482283.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14089/HighFidelity-Beta-Interface-PR13246-a482283.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14089/HighFidelity-Beta-PR13246-a482283.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14089/HighFidelity-Beta-Interface-PR13246-a482283.exe

Zvork commented:
build this please

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14239/HighFidelity-Beta-PR13246-2972267.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14239/HighFidelity-Beta-Interface-PR13246-2972267.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14239/HighFidelity-Beta-PR13246-2972267.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14239/HighFidelity-Beta-Interface-PR13246-2972267.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14349/HighFidelity-Beta-PR13246-5505c59.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14349/HighFidelity-Beta-Interface-PR13246-5505c59.dmg

qaAdamSmith82 commented:
Currently Testing

qaAdamSmith82 commented:
Fail @conklin94122 
I pulled the expected image for this test fromhttps://github.com/Zvork/hifi_tests/blob/velgbuf/tests/engine/render/antialiasing/testStory.md

And although I cannot run them through the Auto Tester evaluation tool, I think the two images (actual and expected) look different enough to warrant a failure.

Additionally, on this PR the interstitial page text jitters noticeably. Much more so than master.

Repro:
1. Launch High Fidelity PR13246
2. Jump between a few different domains
3. Observe the interstitial texts appears overly jittery
4. Nav to your local host, enter coordinates of /8000,8000,800
5. Run https://raw.githubusercontent.com/Zvork/hifi_tests/velgbuf/tests/engine/render/antialiasing/testStory.js
6. Compare the screenshot taken from the above script and compare it to https://raw.githubusercontent.com/Zvork/hifi_tests/velgbuf/tests/engine/render/antialiasing/ExpectedImage_00000.png
7. Observe results
[hifi-log.txt](https://github.com/highfidelity/hifi/files/2512469/hifi-log.txt)
[hifi-log_2018-10-24_14.53.51.txt](https://github.com/highfidelity/hifi/files/2512470/hifi-log_2018-10-24_14.53.51.txt)
![tests engine render antialiasing 00000](https://user-images.githubusercontent.com/41648433/47463985-b832b100-d79c-11e8-9148-c8b954d91880.png)
![expectedimage_00000](https://user-images.githubusercontent.com/41648433/47463986-b832b100-d79c-11e8-85ea-37e6fb98f4a3.png)

Zvork commented:
Ok, I've updated the test and fixed it. I was assuming that the tests fo AA weren't used. Now they should work. As for the jittering text in between domains, I'm trying to understand what exactly is displayed in those screens to debug this. Could someone help me and point me to the script that does this so that I can have a list of the overlay entities that are displayed? I know theres a text overlay but there are two bizarre "square" black entities that are rendered which ruining the AA effect. I just don't know what they are...

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14424/HighFidelity-Beta-PR13246-7b5f3e5.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14424/HighFidelity-Beta-Interface-PR13246-7b5f3e5.exe

Zvork commented:
build this please

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14426/HighFidelity-Beta-PR13246-7b5f3e5.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14426/HighFidelity-Beta-Interface-PR13246-7b5f3e5.exe

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14541/HighFidelity-Beta-PR13246-5f59a7f.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14541/HighFidelity-Beta-Interface-PR13246-5f59a7f.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14573/HighFidelity-Beta-PR13246-f5de319.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14573/HighFidelity-Beta-Interface-PR13246-f5de319.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14573/HighFidelity-Beta-PR13246-f5de319.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14573/HighFidelity-Beta-Interface-PR13246-f5de319.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14699/HighFidelity-Beta-PR13246-b84ae7c.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14699/HighFidelity-Beta-Interface-PR13246-b84ae7c.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14699/HighFidelity-Beta-PR13246-b84ae7c.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14699/HighFidelity-Beta-Interface-PR13246-b84ae7c.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14704/HighFidelity-Beta-PR13246-ef72b6c.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14704/HighFidelity-Beta-PR13246-ef72b6c.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14704/HighFidelity-Beta-Interface-PR13246-ef72b6c.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14704/HighFidelity-Beta-PR13246-ef72b6c.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14704/HighFidelity-Beta-Interface-PR13246-ef72b6c.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14855/HighFidelity-Beta-PR13246-58b37da.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14855/HighFidelity-Beta-Interface-PR13246-58b37da.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14855/HighFidelity-Beta-PR13246-58b37da.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14855/HighFidelity-Beta-Interface-PR13246-58b37da.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14858/HighFidelity-Beta-PR13246-cd494ce.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14858/HighFidelity-Beta-PR13246-cd494ce.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14858/HighFidelity-Beta-Interface-PR13246-cd494ce.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14858/HighFidelity-Beta-PR13246-cd494ce.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14858/HighFidelity-Beta-Interface-PR13246-cd494ce.exe

SamGondelman commented on libraries/render-utils/src/parabola.slv:
```diff
@@ -27,32 +27,41 @@ LAYOUT_STD140(binding=0) uniform parabolaData {
+    vec4 prevPos = pos;
     const float EPSILON = 0.00001;
     vec4 normal;
+    vec4 prevNormal;
```
What are these?

Zvork replied:
<sigh> That's to compute the previous clip space position, the clip space position of that vertex in the previous frame. That's for the velocity buffer used in TAA. As I see in the shader that the vertex position is computed from the world up, if we are to compute the previous position we need to compute the previous normal, etc...

BUT... I see that I don't use the final prevPos which indicates that I am either stupid or unable to focus. Sorry for that, I'll fix it. _prevPositionCS should be computed from prevPos, just like gl_Position is computed from pos.

SamGondelman replied:
I still don't think this is right.  If prevPos = pos, the only thing changing is the normal, which will not result in the correct position change.

SamGondelman replied:
Or...maybe it doesn't matter because pos is in model space?  I am confused now.

Zvork replied:
This is how I see it: we need to find the previous clip space position of this vertex. This depends both on the previous projection view matrix and the previous model space position. The previous projection view matrix is taken car in the transformModelToMonoClipPos function. The previous model space position depends (from what I see in the shader code) both on the normal and the "pos" value. This previous normal is computed from the worldUp vector, which depends on the view matrix. So the previous normal value depends on the previous view matrix. Hence we need to compute the previous worldUp. So although prevPos = pos at the begining of the shader, if the "parabola acceleration" is near 0, it seems that normal != prevNormal and so prevPos != pos.
To be honest, I didn't really try to understand why we were doing all this. I've just tried to extract what depended on the view / model matrices and deduced what should be computed to get a correct previous clip space position.

Zvork replied:
I hope you have a better understanding of what is the correct computation of the previous clip space position, if this is wrong.

SamGondelman replied:
I think the reason this doesn't work is that velocity/acceleration change from frame to frame, so the computation of the previous position using the current values isn't right.

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14993/HighFidelity-Beta-PR13246-47f0601.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14993/HighFidelity-Beta-Interface-PR13246-47f0601.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14993/HighFidelity-Beta-PR13246-47f0601.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14993/HighFidelity-Beta-Interface-PR13246-47f0601.exe

Zvork commented:
There's not a lot of working domains at the time of these last two commits so I'm not yet really 100% confident about these transparent objects. I'll have to recheck once TheSpot-dev is back up and running again, for instance.

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15063/HighFidelity-Beta-PR13246-d32ca78.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15063/HighFidelity-Beta-PR13246-d32ca78.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15063/HighFidelity-Beta-Interface-PR13246-d32ca78.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15063/HighFidelity-Beta-PR13246-d32ca78.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15063/HighFidelity-Beta-Interface-PR13246-d32ca78.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15141/HighFidelity-Beta-PR13246-905c867.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15141/HighFidelity-Beta-PR13246-905c867.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15141/HighFidelity-Beta-Interface-PR13246-905c867.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15141/HighFidelity-Beta-PR13246-905c867.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15141/HighFidelity-Beta-Interface-PR13246-905c867.exe

SamGondelman commented on libraries/render-utils/src/DeferredTransform.slh:
```diff
@@ -16,112 +16,103 @@
+}
+
+int getStereoSideFromUV(float uPos) {
+    return int(uPos >= 0.5 && isStereo()) & 1;
```
These &s don't seem necessary.

Zvork replied:
You're right. We could write it as:
`return int(uPos >= 0.5) & int(isStereo());`
Is that what you mean?

Zvork replied:
or just plain
`return int(uPos >= 0.5 && isStereo());`

Zvork replied:
It's just a habit I took in C++ to & it with 1 to be sure we get 0 or 1, but in GLSL casting bool to int gives either 0 or 1, so we should be safe. I learned something.

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15188/HighFidelity-Beta-PR13246-5be4e1c.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15188/HighFidelity-Beta-PR13246-5be4e1c.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15188/HighFidelity-Beta-Interface-PR13246-5be4e1c.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15188/HighFidelity-Beta-PR13246-5be4e1c.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15188/HighFidelity-Beta-Interface-PR13246-5be4e1c.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15212/HighFidelity-Beta-PR13246-76c9cbe.apk

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15212/HighFidelity-Beta-PR13246-76c9cbe.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15212/HighFidelity-Beta-Interface-PR13246-76c9cbe.exe

Zvork commented:
buid this please

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15213/HighFidelity-Beta-PR13246-76c9cbe.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15213/HighFidelity-Beta-PR13246-76c9cbe.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15213/HighFidelity-Beta-Interface-PR13246-76c9cbe.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15213/HighFidelity-Beta-PR13246-76c9cbe.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15213/HighFidelity-Beta-Interface-PR13246-76c9cbe.exe

samcake commented:
Build this please

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15375/HighFidelity-Beta-PR13246-76c9cbe.apk

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15375/HighFidelity-Beta-PR13246-76c9cbe.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15375/HighFidelity-Beta-Interface-PR13246-76c9cbe.exe

samcake commented:
build this please

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15428/HighFidelity-Beta-PR13246-76c9cbe.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15428/HighFidelity-Beta-Interface-PR13246-76c9cbe.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15428/HighFidelity-Beta-PR13246-76c9cbe.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15428/HighFidelity-Beta-Interface-PR13246-76c9cbe.exe

SamGondelman commented on libraries/render-utils/src/BloomThreshold.shared.slh:
```diff
@@ -8,6 +8,7 @@
+    float _offset;
     float _threshold;
     int _sampleCount;
 };
```
This struct isn't aligned anymore.  And if the offset can be derived from the sample count, why not just do it in the shader?

SamGondelman replied:
I would agree...except the computation is just `(1 - _sampleCount) * 0.5`, right?

Zvork replied:
Yes. I like it when it is lean :-)

SamGondelman commented on libraries/render-utils/src/GeometryCache.h:
```diff
@@ -471,15 +473,26 @@ class GeometryCache : public Dependency {
     static render::ShapePipelinePointer _simpleOpaqueFadePipeline;
     static render::ShapePipelinePointer _simpleTransparentFadePipeline;
     static render::ShapePipelinePointer _simpleWirePipeline;
+    static render::ShapePipelinePointer _simpleWireForwardPipeline;
```
We should store these all in a map so we can just look them up.

Zvork replied:
Oh my god, that would be heaven. Basically we would be doing the shape plumber mechanism over again. So I've postponed this as, as you've seen, this PR is huge and that would be another big refactoring. We've discussed this with @samcake a bit. This my strong opinion that all this GeometryCache shading management nonsense be refactored using the same "plumber" mechanism as the Models.

SamGondelman commented on libraries/render-utils/src/Highlight_aabox.slv:
```diff
@@ -12,10 +12,10 @@
 //
 
-<@include gpu/ShaderConstants.h@>
+<@if not GPU_TRANSFORM_STATE_SLH@>
```
This if should be inside of Transform, not here.

Zvork replied:
yep

SamGondelman commented on libraries/render-utils/src/RenderCommonTask.cpp:
```diff
@@ -72,28 +85,35 @@ void DrawOverlay3D::run(const RenderContextPointer& renderContext, const Inputs&
-            batch.setProjectionTransform(projMat);
-            batch.setProjectionJitter(jitter.x, jitter.y);
-            batch.setViewTransform(viewMat);
+			batch.setProjectionJitterEnabled(_isJitterEnabled);
```
tabs

SamGondelman commented on libraries/render-utils/src/RenderDeferredTask.cpp:
```diff
@@ -256,13 +262,15 @@ void RenderDeferredTask::build(JobModel& task, const render::Varying& input, ren
     const auto toneMappingInputs = ToneMappingDeferred::Inputs(lightingFramebuffer, scaledPrimaryFramebuffer).asVarying();
     task.addJob<ToneMappingDeferred>("ToneMapping", toneMappingInputs);
 
+    const auto toneAndPostRangeTimer = task.addJob<BeginGPURangeTimer>("BeginToneAndPostRangeTimer", "PostToneOverlaysAntialiasing");
```
This timer isn't measuring "Tone and Post" anymore.

Zvork replied:
True. I'll move it to include AA, Bloom and Tonemapping. Is that ok?

SamGondelman commented on libraries/render-utils/src/forward_model.slf:
```diff
@@ -10,74 +10,6 @@
-<@include graphics/Material.slh@>
-<@include graphics/MaterialTextures.slh@>
-<@include render-utils/ShaderConstants.h@>
+<@include forward_model.slh@>
```
I'm in favor of combining our shaders in general...but I think this is not the best PR to do it.  For the sake of the diff, it would be easier to just handle the velocity in each shader.

SamGondelman commented on tools/scribe/src/TextTemplate.cpp:
```diff
@@ -0,0 +1,1016 @@
+    for (const auto& include : _includeFullPaths) {
+        std::cout << include << std::endl;
+    }
+}
```
What is this stuff?

Zvork commented on interface/src/ui/overlays/Cube3DOverlay.cpp:
```diff
@@ -57,15 +57,20 @@ void Cube3DOverlay::render(RenderArgs* args) {
+            } else {
+                shapePipeline = _isSolid ? geometryCache->getOpaqueShapePipeline() : geometryCache->getWireShapePipeline();
+            }
         }
```
Yes, that's true.

Zvork commented on interface/src/ui/overlays/Circle3DOverlay.cpp:
```diff
@@ -80,8 +80,9 @@ void Circle3DOverlay::render(RenderArgs* args) {
 
     Q_ASSERT(args->_batch);
     auto& batch = *args->_batch;
+    const auto isForward = render::ShapeKey(args->_globalShapeKey).isForward();
```
Would you prefer a flag in the RenderArgs, maybe? I hesitated...

SamGondelman replied:
I don't really see the need for this either way...the current system was working fine (until we decide to allow per-object control of the rendering path, which we should do separately).

Zvork replied:
Well, the problem is that I came across some overlay entity classes that were both rendered in the deferred and forward pipeline, so we need a way to determine what shader to use because they do not output the same things at the same place when it is deferred or forward. My opinion is that entities should NOT do any direct batch rendering calls. They shouldn't care if we are in forward, render, shadow or whatever. This is an ongoing discussion with @samcake . But in the meantime, I haven't found another way to handle this (very dirty) case. Maybe there is something wrong somewhere and I am fixing it in the wrong manner, though.

Zvork commented on interface/src/ui/overlays/Text3DOverlay.cpp:
```diff
@@ -104,7 +104,7 @@ void Text3DOverlay::render(RenderArgs* args) {
     glm::vec3 topLeft(-halfDimensions.x, -halfDimensions.y, SLIGHTLY_BEHIND);
     glm::vec3 bottomRight(halfDimensions.x, halfDimensions.y, SLIGHTLY_BEHIND);
-    DependencyManager::get<GeometryCache>()->bindSimpleProgram(batch, false, quadColor.a < 1.0f, false, false, false);
+    DependencyManager::get<GeometryCache>()->bindSimpleProgram(batch, false, quadColor.a < 1.0f, false, false, false, true, true);
```
Yes, we could make a test based on the global shape key...

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15554/HighFidelity-Beta-PR13246-d36c05b.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15554/HighFidelity-Beta-PR13246-d36c05b.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15554/HighFidelity-Beta-Interface-PR13246-d36c05b.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15554/HighFidelity-Beta-PR13246-d36c05b.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15554/HighFidelity-Beta-Interface-PR13246-d36c05b.exe

hifi-gustavo commented:
Can someone from High Fidelity approve this PR?

samcake commented:
WE wil reopen when appropriate
